rasa/utils/tensorflow/model_data.py

```diff
@@ -843,7 +939,39 @@ def _pad_dense_data(array_of_dense: np.ndarray) -> np.ndarray:
         return data_padded.astype(np.float32)
 
     @staticmethod
-    def _scipy_matrix_to_values(array_of_sparse: np.ndarray) -> List[np.ndarray]:
+    def _pad_4d_dense_data(array_of_array_of_dense: FeatureArray) -> np.ndarray:
```

Comment:
ragged tensors are not suitable for us?

Reply:
You are too fast with reviewing :smile:  It is not yet fully working, wanted to look into that once everything else works. Please be patient :slightly_smiling_face:

Reply:
oh sorry 😅 was excited to look how subclassing ndarray idea worked out

Reply:
@Ghostvv `RaggedTensors` look good, however they are currently not supported by `tf.data.Dataset.from_generator`... It is already fixed (https://github.com/tensorflow/tensorflow/pull/41981), but seem to be not yet released. We would also need to update some of our layers I guess. I'll create an issue for it to update it in the future.

Reply:
https://github.com/RasaHQ/rasa/issues/6909